File: app/src/main/java/com/example/mpesademo/Interceptor/AccessTokenInterceptor.java
package com.example.mpesademo.Interceptor;

import android.util.Base64;

import com.example.mpesademo.BuildConfig;

import java.io.IOException;

import androidx.annotation.NonNull;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AccessTokenInterceptor implements Interceptor {
    public AccessTokenInterceptor() {

    }

    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {


        String keys = com.example.mpesademo.BuildConfig.CONSUMER_KEY + ":" + BuildConfig.CONSUMER_SECRET;
        Request request = chain.request().newBuilder()
                .addHeader("Authorization", "Basic " + Base64.encodeToString(keys.getBytes(), Base64.NO_WRAP))
                .build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/example/mpesademo/interceptor/AccessTokenInterceptor.kt
package com.example.mpesademo.interceptor;

import android.util.Base64;

import com.example.mpesademo.BuildConfig;
import com.example.mpesademo.Constants
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

class AccessTokenInterceptor: Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val keys:String = Constants.DARAJA_CONSUMER_KEY + ":" + Constants.DARAJA_CONSUMER_SECRET;
        val request: Request = chain.request().newBuilder()
            .addHeader("Authorization", "Basic " + Base64.encodeToString(keys.toByteArray(), Base64.NO_WRAP))
            .build();
        return chain.proceed(request);
    }
}
